translate/storage/lisa.py

```diff
@@ -329,20 +333,40 @@ def serialize_hook(treestring):
     def serialize(self, out=None):
         """Converts to a string containing the file's XML"""
         root = self.document.getroot()
+        xml_quote_format = "'"
+        xml_encoding = self.encoding.lower()
+
         if self.XMLdoublequotes:
-            if self.XMLuppercaseEncoding:
-                out.write(b'<?xml version="1.0" encoding="UTF-8"?>\n')
-            else:
-                out.write(b'<?xml version="1.0" encoding="utf-8"?>\n')
+            xml_quote_format = '"'
+
+        if self.XMLuppercaseEncoding:
+            xml_encoding = self.encoding.upper()
+
+        xml_declaration = f"<?xml version={xml_quote_format}1.0{xml_quote_format} encoding={xml_quote_format}{xml_encoding}{xml_quote_format}?>\n"
+
+        out.write(xml_declaration.encode(self.encoding))
+
         if self.XMLindent:
             reindent(root, **self.XMLindent)
-        treestring = etree.tostring(
-            self.document,
-            pretty_print=not self.XMLindent,
-            xml_declaration=not self.XMLdoublequotes,
-            encoding=self.encoding,
-            doctype=self.XMLdoctype,
-        )
+        # When XMLSelfClosingTags flag is active, relay
+        # against LXML library - otherwise we're using
+        # standard xml library.
+        if self.XMLSelfClosingTags:
+            treestring = etree.tostring(
+                self.document,
+                pretty_print=not self.XMLindent,
+                encoding=self.encoding,
+                doctype=self.XMLdoctype,
+            )
+        else:
+            if self.XMLdoctype:
+                out.write(b"" + self.XMLdoctype + "\n")
+            ElementTree.register_namespace("", self.namespace)
+            treestring = ElementTree.tostring(
+                root,
+                encoding=self.encoding,
+                short_empty_elements=False,
+            )
```

Comment:
Actually `default_namespace` is present in `write` for all supported Python versions, and `tostring` is just a thin wrapper around it, so we could use `write` directly as well.

Something like this (not tested at all):

```suggestion
            stream = io.BytesIO()
            ElementTree(root).write(stream, self.encoding, short_empty_elements=False)
            treestring = stream.getvalue()
```